tests/conftest.py

```
@@ -53,3 +58,20 @@ def ansible_no_error(request):
     assert request.ansible_process.returncode == 0
 
 
+@when(parsers.parse("I run 'kubectl {kubectl_args}' in a supported shell"))
+def run_kubectl_command(request, kubeconfig, kubectl_args):
+    kubectl_command = "kubectl {}".format(kubectl_args)
+    kubectl_run = run_make_shell(
+        kubectl_command,
+        env={'KUBECONFIG': kubeconfig},
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    if kubectl_run.returncode == 0:
+        request.kubectl_result = kubectl_run
+        return kubectl_run
+    else:
+        logging.error("kubectl stdout: {}\n stderr: {}".format(
```

Shouldn't this raise instead of implicitly returning `None`?

---

But some test might require to test error of kubectl (delete a locked pv, or whatever). But indeed this in not ideal (mainly because I don't return the kubectl_run in case of error.)